src/import_edf.jl

```diff
@@ -2,32 +2,22 @@
 ##### `EDF.Signal` label handling
 #####
 
-function edf_type_and_spec(label::AbstractString)
-    parsed = split(label; limit=2, keepempty=false)
-    if length(parsed) == 2
-        type = replace(parsed[1], r"\s"=>"")
-        spec = replace(parsed[2], r"\s"=>"")
-    else
-        # if no clear specification is present, the whole label
-        # is just considered the "type" by the EDF+ standard
-        type = replace(label, r"\s"=>"")
-        spec = nothing
-    end
-    return type, spec
-end
-
 # This function:
 # - ensures the given label is whitespace-stripped, lowercase, and parens-free
 # - strips trailing generic EDF references (e.g. "ref", "ref2", etc.)
 # - replaces all references with the appropriate name as specified by `canonical_names`
 # - replaces `+` with `_plus_` and `/` with `_over_`
-# - returns the initial reference name (w/o prefix sign, if present) and the entire label
+# - returns the initial reference name (w/o prefix sign, if present) and the entire label;
+#   the initial reference name should match the canonical channel name,
+#   otherwise the channel extraction will be rejected.
 function _normalize_references(original_label, canonical_names)
-    label = replace(lowercase(original_label), r"\s"=>"")
+    label = replace(_safe_lowercase(original_label), r"\s"=>"")
     label = replace(replace(label, '('=>""), ')'=>"")
+    label = replace(label, r"\*$"=>"")
     label = replace(label, '-'=>'…')
     label = replace(label, '+'=>"…+…")
     label = replace(label, '/'=>"…/…")
+    label = !isnothing(match(r"^\[.*\]$", label)) ? label[2:end-1] : label
```

Comment:
```suggestion
    m = match(r"^\[(.*)\]$", label)
    if m !== nothing
        label = only(match.captures)
    end
```
This is a little easier to follow, IMO.

Reply:
ooh `only`